### DISCUSSION TITLE
How to prevent Foundry from recompiling all contracts after a small change in a test file?

### DISCUSSION THREAD

**Discussion by rspadinger** [foundry-rs/foundry#6950] [Help]
I have a large project with over 50 smart contracts and test files. Whenever I make a small change in one single test file, all (or almost all) .sol files get recompiled, which takes several minutes. This makes testing very arduous. I don't have this issue, when I execute forge build - only the modified smart contract will be recompiled, which is rather quick. However, when I run forge test... everything gets recompiled. Is there a way to compile only the test file I'm currently working on and to skip compilation for all other .sol files?

**Comment by DaniPopes** [comment]
I can't really reproduce this, if you feel like it's a problem please open an issue with more details.

In the meantime, you can filter input files and tests with `match` options, see `forge test --help`.
